Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -1587,4 +1590,235 @@ void Node::build_accessibility_tree(AccessibilityTreeNode& parent) const
+        return String::from_deprecated_string(text_node->data());
+    }
+    // TODO: H. Otherwise, if the current node is a descendant of an element whose Accessible Name or Accessible Description is being computed, and contains descendants, proceed to 2F.i.
+
+    // I. Otherwise, if the current node has a Tooltip attribute, return its value.
+    // https://www.w3.org/TR/accname-1.2/#dfn-tooltip-attribute
+    // Any host language attribute that would result in a user agent generating a tooltip such as in response to a mouse hover in desktop user agents.
+    // FIXME: Support SVG tooltips and CSS tooltips
+    if (is<HTML::HTMLElement>(this)) {
+        auto const* element = static_cast<HTML::HTMLElement const*>(this);
+        auto tooltip = element->title();
+        if (!tooltip.is_empty() && !tooltip.is_null())
+            return String::from_deprecated_string(tooltip);
+    }
+    // Append the result of each step above, with a space, to the total accumulated text.
+    //
+    // After all steps are completed, the total accumulated text is used as the accessible name or accessible description of the element that initiated the computation.
+    return total_accumulated_text.to_string();
+}
+
+// https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_name
+ErrorOr<String> Node::accessible_name(Document const& document) const
+{
+    HashTable<i32> visited_nodes;
+    // User agents MUST compute an accessible name using the rules outlined below in the section titled Accessible Name and Description Computation.
+    return name_or_description(NameOrDescription::Name, document, visited_nodes);
+}
+
+// https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_description
+ErrorOr<String> Node::accessible_description(Document const& document) const
+{
+    // If aria-describedby is present, user agents MUST compute the accessible description by concatenating the text alternatives for elements referenced by an aria-describedby attribute on the current element.
+    // The text alternatives for the referenced elements are computed using a number of methods, outlined below in the section titled Accessible Name and Description Computation.
+    if (is_element()) {
+        HashTable<i32> visited_nodes;
+        StringBuilder builder;
+        auto const* element = static_cast<Element const*>(this);
+        auto id_list = element->aria_described_by().split_view(Infra::is_ascii_whitespace);
+        for (auto const& id : id_list) {
+            if (auto description_element = document.get_element_by_id(id)) {
+                auto description = TRY(description_element->name_or_description(NameOrDescription::Description, document, visited_nodes));
+                if (!description.is_empty()) {
+                    if (builder.is_empty()) {
+                        builder.append(description);
+                    } else {
+                        builder.append(" "sv);
+                        builder.append(description);
```

Comment:
Again this function returns `ErrorOr` so we should be able to use `try_append()` instead here (a little verbose, but seems silly not to)